src/main/java/com/powsybl/openloadflow/util/Reports.java

```diff
@@ -242,26 +244,49 @@ public static Reporter createNewtonRaphsonMismatchReporter(Reporter reporter, in
-                    .withDefaultMessage("Initial mismatch" + mismatchDetails);
-        } else {
-            reportBuilder.withKey("NRIterationMismatch")
-                    .withDefaultMessage("Iteration ${iteration} mismatch" + mismatchDetails)
-                    .withValue(ITERATION, iteration);
-        }
-
-        reporter.report(reportBuilder.withValue("equationType", acEquationType)
-                .withTypedValue("mismatch", mismatch, OpenLoadFlowReportConstants.MISMATCH_TYPED_VALUE)
-                .withValue("busId", busId)
-                .withTypedValue("busV", busV, TypedValue.VOLTAGE)
-                .withTypedValue("busPhi", busPhi, TypedValue.ANGLE)
-                .withSeverity(TypedValue.TRACE_SEVERITY)
-                .build());
+    public static void reportNewtonRaphsonMismatch(Reporter reporter, String acEquationType, double mismatch, int iteration, NewtonRaphson.NRmismatchBusInfo nRmismatchBusInfo) {
+        Map<String, TypedValue> subReporterMap = new HashMap<>();
+        subReporterMap.put("equationType", new TypedValue(acEquationType, "String"));
+        subReporterMap.put("mismatch", new TypedValue(mismatch, OpenLoadFlowReportConstants.MISMATCH_TYPED_VALUE));
+
+        Reporter subReporter = reporter.createSubReporter(iteration == -1 ? "NRInitialMismatch" : "NRIterationMismatch", "Mismatch on ${equationType}: ${mismatch}", subReporterMap);
+
+        ReportBuilder busIdReportBuilder = Report.builder();
+        busIdReportBuilder.withKey("NRMismatchBusId")
+                .withDefaultMessage("Bus       Id       : ${busId}")
+                .withValue("busId", nRmismatchBusInfo.busId());
+
+        ReportBuilder busNominalVReportBuilder = Report.builder();
+        busNominalVReportBuilder.withKey("NRMismatchBusNominalV")
+                .withDefaultMessage("Bus nominalV [  kV]: ${busNominalV}")
+                .withValue("busNominalV", nRmismatchBusInfo.busNominalV());
+
+        ReportBuilder busVReportBuilder = Report.builder();
+        busVReportBuilder.withKey("NRMismatchBusV")
+                .withDefaultMessage("Bus        V [p.u.]: ${busV}")
+                .withValue("busV", nRmismatchBusInfo.busV());
+
+        ReportBuilder busPhiReportBuilder = Report.builder();
+        busPhiReportBuilder.withKey("NRMismatchBusPhi")
+                .withDefaultMessage("Bus      Phi [ rad]: ${busPhi}")
+                .withValue("busPhi", nRmismatchBusInfo.busPhi());
+
+        ReportBuilder busPReportBuilder = Report.builder();
+        busPReportBuilder.withKey("NRMismatchBusSumP")
+                .withDefaultMessage("Bus     sumP [  MW]: ${busSumP}")
+                .withValue("busSumP", nRmismatchBusInfo.busSumP());
+
+        ReportBuilder busQReportBuilder = Report.builder();
+        busQReportBuilder.withKey("NRMismatchBusSumQ")
+                .withDefaultMessage("Bus     sumQ [MVar]: ${busSumQ}")
+                .withValue("busSumQ", nRmismatchBusInfo.busSumQ());
```

Comment:
@geofjamg @annetill your advice ?

Although padding is ok for logs in `src/main/java/com/powsybl/openloadflow/ac/solver/NewtonRaphson.java`, I am not confortable doing the same for reports in the default messages. Here it looks very pretty if using powsybl-core's ReporterModel and its printing method, but keep in mind that reporting is just an API, so here you are making things looking nice for a particular/specific Reporter implementation.

I would prefer to remove all these space paddings here. OLF users can provide their own Reporter implementation that will format the way they want the raw values.

Reply:
Agreed, just one space.